cachi2/core/package_managers/pip.py

```diff
@@ -1891,18 +1891,9 @@ def _is_valid(pkg: pypi_simple.DistributionPackage) -> bool:
 
     if sdists:
         best_sdist = max(sdists, key=_sdist_preference)
+        processed_dpis.append(best_sdist)
         if best_sdist.is_yanked:
-            raise PackageRejected(
-                f"All sdists for package {name}=={version} are yanked",
-                solution=(
-                    f"Please update the {name} version in your requirements file.\n"
-                    "Usually, when a version gets yanked from PyPI, there will already "
-                    "be a fixed version available.\n"
-                    "Otherwise, you may need to pin to the previous version."
-                ),
-            )
-        if best_sdist:
-            processed_dpis.append(best_sdist)
+            log.warning("All sdists for package %s==%s are yanked", name, version)
```

Comment:
**Should** we keep a warning in here along the lines of "Please update the version in your requirements file to a current version"?

Reply:
> It's possible that the deps in question weren't pinned, though (e.g. '==' or '==='), which seems to be prereq according to the [applicable PEP](https://peps.python.org/pep-0592/#installers).

I installed a random yanked package:

```
$ pip install terminado==0.13.0

Collecting terminado==0.13.0
  Downloading terminado-0.13.0-py3-none-any.whl.metadata (761 bytes)
Requirement already satisfied: tornado>=4 in ./venv/lib64/python3.12/site-packages (from terminado==0.13.0) (6.4)
Requirement already satisfied: ptyprocess in ./venv/lib64/python3.12/site-packages (from terminado==0.13.0) (0.7.0)
WARNING: The candidate selected for download or install is a yanked version: 'terminado' candidate (version 0.13.0 at https://files.pythonhosted.org/packages/02/6c/c3b012d026678d5cac6d4e26d6857145e22b94ccef195e7c483c3f6c3731/terminado-0.13.0-py3-none-any.whl (from https://pypi.org/simple/terminado/))
Reason for being yanked: Missing python_requires metadata
Downloading terminado-0.13.0-py3-none-any.whl (14 kB)
Installing collected packages: terminado
  Attempting uninstall: terminado
    Found existing installation: terminado 0.8.3
    Uninstalling terminado-0.8.3:
      Successfully uninstalled terminado-0.8.3
Successfully installed terminado-0.13.0
```